src/pages/docs/tutorials/tutorial-typescript-enact/component-with-ts-enact/index.md

```diff
@@ -22,60 +22,84 @@ const CounterBase = kind({
         count: 0
     },
 
-    render: ({onIncrementClick, onDecrementClick, onResetClick, count, ...rest}) => (
+    render: ({count}) => (
         <div>
             <h1>{count}</h1>
-            <Button onClick={onDecrementClick}>Decrement --</Button>
-            <Button onClick={onResetClick}>Reset</Button>
-            <Button onClick={onIncrementClick}>Increment ++</Button>
+            <Button>Decrement --</Button>
+            <Button>Reset</Button>
+            <Button>Increment ++</Button>
         </div>
     )
 });
 
 export default CounterBase;
 ```
 
-### View the Counter in the Browser
-
-![Enact Typescript Counter](Typescript_Enact_view.png)
+The above code holds the definition of the `Counter` component. 
 
-The above code holds the definition of the `Counter` component. Using `kind()` we are binding `defaultProps`, `render` and all the events together.
+Now lets add events on the buttons using `kind()` we are binding `defaultProps`, `render` and all the events together.
 
 > Please check API documentation in the Core Library to know more about [kind](../../../modules/core/kind/) and [handle](../../../modules/core/handle/)
 
-We will add `handler` inside the `kind` declaration to handle the click event on the buttons:
+Update the render block with `onClick` event to use the `handler` functions. Add a `handlers` block inside the kind declaration:
 
 ```ts
+//Counter.tsx
+
+import {adaptEvent, forward, handler} from '@enact/core/handle';
+import Button from '@enact/moonstone/Button';
+import kind from '@enact/core/kind';
+import React from 'react';
+
+const CounterBase = kind({
+    name: 'Counter',
+
+    defaultProps: {
+        count: 0
+    },
+
     handlers: {
-        onDecrementClick: createHandler(count => count - 1),
-        onIncrementClick: createHandler(count => count + 1),
-        onResetClick: createHandler(() => 0)
+        onDecrementClick: count => count - 1,
+        onIncrementClick: count => count + 1,
+        onResetClick: count => 0
```

Comment:
Should we have this here?  Perhaps show a different approach since this doesn't work?  Perhaps show not using handlers and just calling the passed-in event handler?